build-on-push/platform_ci/platform_ci/brew.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2016 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module provides multiple classes representing Brew entities, mostly
+used to issue testing (scratch) builds and evaluating results of such
+requests.
+"""
+
+import os.path
+import logging
+import subprocess
+import tempfile
+import platform_ci.notifications as notifications
+
+
+class BuildToCommitterMapping(object):
+    """Provides a simple way to store committer addresses for CI-issued builds.
+
+    When CI issues Brew builds automatically, it does so under its own machine
+    credentials, not under credentials of the person who actually performed the
+    action that triggered the CI action (usually a git push). Therefore, Brew
+    tracks the CI machine account as a Brew build issuer, which may cause
+    later notifications (e.g. when such build is tested later) to be sent to
+    the machine account address instead of the right recipient.
+
+    This class allows to store committer information at the time when the build
+    is issued, mapping the issued build task ID to this committer information,
+    so it can be retrieved later.
+
+    Currently, the class stores the mapping in the filesystem, so it relies on
+    the later processing happening on the same Jenkins slave. This is fragile
+    and should be improved.
```

Comment:
"To minimize the delay between a push and a build, it is recommended to set up the post-receive hook in the repository to poke Jenkins when a new push occurs." https://wiki.jenkins-ci.org/display/JENKINS/Git+Plugin#GitPlugin-Pushnotificationfromrepository 

Did we explore or do a POC with git hooks ? Could you share the findings (if any)?